native/KotlinIOS/KotlinIOS/ViewController.swift

```diff
@@ -1,20 +1,113 @@
 import UIKit
 import SharedCode
 
-class ViewController: UIViewController {
+class TableViewCell: UITableViewCell {
+    
+    @IBOutlet weak var departureLabel: UILabel!
+    @IBOutlet weak var arrivalLabel: UILabel!
+}
 
-    @IBOutlet private var label: UILabel!
+class ViewController: UIViewController, ApplicationContractView{
+
+    @IBOutlet weak var dropDown1: UIPickerView!
+    @IBOutlet weak var dropDown2: UIPickerView!
+    @IBOutlet weak var tableView: UITableView!
+    
+    private var tableData: NSMutableArray = NSMutableArray()
+    
+    private var stations: [String] = [String]()
+    
 
     private let presenter: ApplicationContractPresenter = ApplicationPresenter()
     
     override func viewDidLoad() {
         super.viewDidLoad()
         presenter.onViewTaken(view: self)
+        
+        setUpPickers()
+        setUpTable()
+    }
+    
+    
+    @IBAction func onButtonTapped(){
+        let originStation = stations[dropDown1.selectedRow(inComponent:0)]
+        let finalStation = stations[dropDown2.selectedRow(inComponent:0)]
+        presenter.makeTrainSearch(originCrs: originStation,destinationCrs:  finalStation)
+        
+    }
+}
+
+extension ViewController {
+    func setStationNames(stationNames: Array<String>) {
+        stations = stationNames
+    }
+}
+
+extension ViewController: UIPickerViewDelegate, UIPickerViewDataSource{
+    
+    func setUpPickers() {
+        self.dropDown1.delegate = self
+        self.dropDown1.dataSource = self
+        
+        self.dropDown2.delegate = self
+        self.dropDown2.dataSource = self
+    }
+    
+    func numberOfComponents(in pickerView: UIPickerView) -> Int {
+        return 1
+    }
+    
+    func pickerView(_ pickerView:UIPickerView, numberOfRowsInComponent component:Int)-> Int{
+        return stations.count
+    }
+    
+    func pickerView(_ pickerView: UIPickerView, titleForRow row: Int, forComponent component: Int)->String?{
+        return stations[row]
+    }
+}
+
+
+extension ViewController: UITableViewDataSource, UITableViewDelegate {
+    
+    func setUpTable(){
+        self.tableView.delegate = self
+        self.tableView.dataSource = self
+        
+        //tableData = ["STRING", "STRING"]
+        registerTableViewCells()
+    }
+    
+    func registerTableViewCells() {
+        let trainCell = UINib(nibName: "TableViewCell", bundle: nil)
+        self.tableView.register(trainCell, forCellReuseIdentifier: "TableViewCell")
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) ->
+        UITableViewCell {
+        if let cell = tableView.dequeueReusableCell(withIdentifier: "TableViewCell", for: indexPath) as? TableViewCell{
```

Comment:
Might be nice to pull "TableViewCell" out as a constant in the file as we use a few times